zk/conn.go

```diff
@@ -200,6 +204,8 @@ func Connect(servers []string, sessionTimeout time.Duration, options ...connOpti
 		logger:       DefaultLogger,
 		bufferSize:   defaultBufferSize,
 
+		reconnectOnSessionExpiration: true,
```

Comment:
Normally in Go you want the default values of a struct to be correct without initialization such that they maintain status quo.

The original implementation is wrong, so that means rephrasing this as `closeOnSessionExpiration` which defaults to `false` unless we override it.